source/MaterialXCore/Document.h

```diff
@@ -339,12 +339,15 @@ class MX_CORE_API Document : public GraphElement
     /// @param node Node type for the new declaration
     /// @param version Version for the new declaration
     /// @param isDefaultVersion If a version is specified is thie definition the default version
+    /// @param nodeGroup Optional node group for the new declaration. The Default value is an emptry string.
     /// @param newGraphName Make a copy of this NodeGraph with the given name if a non-empty name is provided. Otherwise
     ///        modify the existing NodeGraph. Default value is an empty string.
-    /// @param nodeGroup Optional node group for the new declaration. The Default value is an emptry string.
+    /// @param docString Optional documentation string. Default is an empty string.
+    /// @param namesSpace Optional namespace for the Nodedef and NodeGraph. Default is an empty string.
     /// @return New declaration if successful.
     NodeDefPtr addNodeDefFromGraph(const NodeGraphPtr nodeGraph, const string& nodeDefName, const string& node, const string& version,
```

Comment:
I'm a bit wary about the complexity of this function signature, and I wonder if it could be streamlined in this changelist, rather than making it more all-encompassing.  For arguments such as `docString` and `nodeGroup`, could these instead be assigned outside of the function, since the caller has access to the returned `NodeDefPtr`?  If it's possible to remove all of the function arguments that can easily be set by the caller, then I think this would be a better approach, leaving the helper function to focus on the core logic of creating a NodeDef from a NodeGraph.